src/app/zap-templates/zcl/pump-configuration-and-control-cluster.xml

```diff
@@ -0,0 +1,101 @@
+    <attribute side="server" code="0x000A" define="MAX_CONST_FLOW" type="INT16U" min="0x0000" max="0xFFFF" writable="false" optional="true">max const flow</attribute>
+    <attribute side="server" code="0x000B" define="MIN_CONST_TEMP" type="INT16S" min="0x0000" max="0xFFFF" writable="false" optional="true">min const temp</attribute>
+    <attribute side="server" code="0x000C" define="MAX_CONST_TEMP" type="INT16S" min="0x0000" max="0x7FFF" writable="false" optional="true">max const temp</attribute>
+    <attribute side="server" code="0x0010" define="PUMP_STATUS" type="BITMAP16" writable="false" reportable="true" optional="true">pump status</attribute>
+    <attribute side="server" code="0x0011" define="EFFECTIVE_OPERATION_MODE" type="ENUM8" min="0x00" max="0xFE" writable="false" optional="false">effective operation mode</attribute>
+    <attribute side="server" code="0x0012" define="EFFECTIVE_CONTROL_MODE" type="ENUM8" min="0x00" max="0xFE" writable="false" optional="false">effective control mode</attribute>
+    <attribute side="server" code="0x0013" define="CAPACITY" type="INT16S" min="0x0000" max="0x7FFF" writable="false" reportable="true" optional="false">capacity</attribute>
+    <attribute side="server" code="0x0014" define="SPEED" type="INT16U" min="0x0000" max="0xFFFE" writable="false" optional="true">speed</attribute>
+    <attribute side="server" code="0x0015" define="LIFETIME_RUNNING_HOURS" type="INT24U" min="0x000000" max="0xFFFFFE" writable="true" default="0x000000" optional="true">lifetime running hours</attribute>
+    <attribute side="server" code="0x0016" define="PUMP_POWER" type="INT24U" min="0x000000" max="0xFFFFFE" writable="true" optional="true">power</attribute>
+    <attribute side="server" code="0x0017" define="LIFETIME_ENERGY_CONSUMED" type="INT32U" min="0x00000000" max="0xFFFFFFFE" writable="false" default="0x00000000" optional="true">lifetime energy consumed</attribute>
+    <attribute side="server" code="0x0020" define="OPERATION_MODE" type="ENUM8" min="0x00" max="0xFE" writable="true" default="0x00" optional="false">operation mode</attribute>
+    <attribute side="server" code="0x0021" define="CONTROL_MODE" type="ENUM8" min="0x00" max="0xFE" writable="true" default="0x00" optional="true">control mode</attribute>
+    <attribute side="server" code="0x0022" define="PUMP_ALARM_MASK" type="BITMAP16" writable="false" optional="true">alarm mask</attribute>
+  </cluster>
+
+  <bitmap name="PumpStatus" type="BITMAP16">
+    <field name="DeviceFault" mask="0x1"/>
+    <field name="Supplyfault" mask="0x2"/>
+    <field name="SpeedLow" mask="0x4"/>
+    <field name="SpeedHigh" mask="0x8"/>
+    <field name="LocalOverride" mask="0x10"/>
+    <field name="Running" mask="0x20"/>
+    <field name="RemotePressure" mask="0x40"/>
+    <field name="RemoteFlow" mask="0x80"/>
+    <field name="remoteTemperature" mask="0x100"/>
+  </bitmap>
+
+  <enum name="OperationMode" type="ENUM8">
```

Comment:
Sorry to not have noticed that first. It would be better to prefix them with `Pump`. I understand that the spec says: `OperationMode`, `ControlMode` and `AlarmMask`, but `AlarmMask` for example is a very common name and currently in the spec, it is used by `DoorLock` and `Thermostat`too.

Reply:
What is the preferred way forward here? 

Raise an Issue against the spec first, and live with these common names here for now?

Or just make the changes here now and let the implementation diverge from the spec, until the spec catches up?

Something else?

Reply:
Let's make the change here, and file an issue against the spec in parallel.

Reply:
Thinking a bit about this, I have a few questions / comments.

What you say is basically that the namespace is flat, is there really no such thing as chip::pump::attribute_1?

If we make this change that presumably mean that every single cluster (and device?) needs to update its attribute and command names...

It seems that these names worked in ZCL, so there was some namespaces there? Also if we make this change it becomes a bit less compatible with ZCL (although the translation would be trivial).

So making this change here should be easy, but the implication is that it will ripple out in the whole spec. So considering this Is this the approach you want to go with? In my personal opinion it would be worthwhile to look at using namespaces and maybe how this was handled (if it was handled) in ZCL.

Reply:
> Thinking a bit about this, I have a few questions / comments.
> 
> What you say is basically that the namespace is flat, is there really no such thing as chip::pump::attribute_1?
> 

Please note that when it comes to clusters a lot of the current code is inherited from a C based implementation. It would be easier to have namespaces as you suggests but ones needs to fix the backend first.

> If we make this change that presumably mean that every single cluster (and device?) needs to update its attribute and command names...
> 
> It seems that these names worked in ZCL, so there was some namespaces there? Also if we make this change it becomes a bit less compatible with ZCL (although the translation would be trivial).

Not really. In ZCL the names where: `PumpStatus`, `PumpControlMode`, `PumpOperationMode`, `PumpAlarmMask`. So in fact, renaming those would match what ZCL was doing.

> 
> So making this change here should be easy, but the implication is that it will ripple out in the whole spec. So considering this Is this the approach you want to go with? In my personal opinion it would be worthwhile to look at using namespaces and maybe how this was handled (if it was handled) in ZCL.

I agree with you about name spacing stuffs. My comment applies for legacy reasons, not my personal opinion.

Reply:
Thanks for the clarification, I'm gonna update as indicated above.

Reply:
So I believe the three types OperationMode, ControlMode and AlarmMask have been appropiately prepended with Pump now, but lets see what the bot says, if there is anything else let me know.

Someone should open an issue on the spec with a reference to this PR.

I note that the naming of atrtibutes and fields is not very stringent, a few of them have their *name* prepended with Pump, but most do not, e.g. we have this kind of thing (inherited from ZCL) which is kind of perplexing :)

```
<field name="MotorTemperatureHigh" mask="0x40"/>
<field name="PumpMotorFatalFailure" mask="0x80"/>
```

Now atleast all type names a prepended with Pump. Personally I like simple rules for these kind of things, so if you want to remove/add Pump from/to all fields and attributes, let me know.